### PR TITLE
[feat] Create/Edit actions on tabs

### PR DESCRIPTION
**Pull request by marcoviappiani**
- Adds Create action to tabs view
- Adds Edit action to tabs view

The above action allow to specify uisref and tab title, but not yet the roles. Roles are being set based on server predefined (admin and owner are the default).
- updates .gitignore to ignore DS_Store

### CODE REVIEW

File: modules/core/client/controllers/addTabDialogue.controller.client.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  vm.title = 'Create Tab';
+
+
+  //////////
+
+  function hide () {
+    $mdDialog.hide();
+  }
+
+  function cancel () {
```
Comment (codydaig) [inline modules/core/client/controllers/addTabDialogue.controller.client.js:22]: Alphabetical Order

File: modules/core/client/controllers/adminTabs.controller.client.js [hunk 2]
```diff
@@ -3,15 +3,17 @@
 angular.module('tropicalbs')
   .controller('AdminTabsController', AdminTabsController);
 
-AdminTabsController.$inject =  ['tabsService'];
+  AdminTabsController.$inject =  ['tabsService', '$mdDialog'];
```
Comment (codydaig) [inline modules/core/client/controllers/adminTabs.controller.client.js]: Remove Space

File: modules/core/client/controllers/adminTabs.controller.client.js [hunk 3]
```diff
@@ -26,7 +28,46 @@ function AdminTabsController (tabsService) {
...
 
+  // function removeTabDialogue () {
+  //   //remove a tab
+  // }
+
   function setAllTabs (res) {
     vm.tabs = res;
   }
+
+  function createTabDialogue (ev, tab) {
```
Comment (codydaig) [inline modules/core/client/controllers/adminTabs.controller.client.js:35]: Alphabetical Order

File: modules/core/client/controllers/adminTabs.controller.client.js [hunk 4]
```diff
@@ -26,7 +28,46 @@ function AdminTabsController (tabsService) {
...
+    $mdDialog.show({
+      controller: 'AddTabDialogueController',
+      controllerAs: 'vm',
+      templateUrl: '../../../../modules/core/client/views/adminTabsDialogue.view.client.html',
+      parent: angular.element(document.body),
+      targetEvent: ev,
+      clickOutsideToClose: true
+    })
+    .then(function(answer) {
```
Comment (codydaig) [inline modules/core/client/controllers/adminTabs.controller.client.js]: Nested Function

File: modules/core/client/controllers/adminTabs.controller.client.js [hunk 5]
```diff
@@ -26,7 +28,46 @@ function AdminTabsController (tabsService) {
...
+      //success TODO
+    }, function() {
```
Comment (codydaig) [inline modules/core/client/controllers/adminTabs.controller.client.js]: Can't this be a catch?

File: modules/core/client/controllers/adminTabs.controller.client.js [hunk 6]
```diff
@@ -26,7 +28,46 @@ function AdminTabsController (tabsService) {
...
+      controllerAs: 'vm',
+      templateUrl: '../../../../modules/core/client/views/adminTabsDialogue.view.client.html',
+      parent: angular.element(document.body),
+      locals: {
+             tab: tab,
+           },
+      targetEvent: ev,
+      clickOutsideToClose: true
+    })
+    .then(function(answer) {
```
Comment (codydaig) [inline modules/core/client/controllers/adminTabs.controller.client.js]: Nested function, same as above.

File: modules/core/client/controllers/editTabDialogue.controller.client.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+angular.module('tropicalbs')
+  .controller('EditTabDialogueController', EditTabDialogueController);
+
+  EditTabDialogueController.$inject =  ['$mdDialog', 'tabsService', 'tab'];
```
Comment (codydaig) [inline modules/core/client/controllers/editTabDialogue.controller.client.js]: Indentation

File: modules/core/client/controllers/editTabDialogue.controller.client.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  function activate () {
+    _.assign(vm.tab, tab);
+    vm.title = 'Edit Tab';
+  }
+
+  function hide () {
+    $mdDialog.hide();
+  }
+
+  function cancel () {
```
Comment (codydaig) [inline modules/core/client/controllers/editTabDialogue.controller.client.js:33]: Alphabetical Order

File: modules/core/client/services/tabs.service.client.js [hunk 9]
```diff
@@ -7,6 +7,10 @@ tabsService.$inject = ['$http'];
 
 function tabsService ($http) {
   var tabs = {
+    createTab: createTab,
+    deleteTab: deleteTab,
+    updateTab: updateTab,
+    getTab: getTab,
```
Comment (codydaig) [inline modules/core/client/services/tabs.service.client.js:13]: Alphabetical Order

File: modules/core/client/views/adminTabsDialogue.view.client.html [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+      </md-input-container>
+      <md-input-container>
+        <label>Ui-sref</label>
+        <input required name="Ui-sref" ng-model="vm.tab.uisref">
+        <!-- <div ng-messages="editUserForm.email.$error">
+          <div ng-message="required">This is required.</div>
+        </div> -->
+      </md-input-container>
+      <!-- <h4>Visible to roles</h4> -->
+<!--       <md-chips ng-model="vm.tab.roles" md-autocomplete-snap md-require-match="true">
```
Comment (codydaig) [inline modules/core/client/views/adminTabsDialogue.view.client.html]: Indentation

File: modules/core/client/controllers/addTabDialogue.controller.client.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+angular.module('tropicalbs')
+  .controller('AddTabDialogueController', AddTabDialogueController);
+
+  AddTabDialogueController.$inject =  ['$mdDialog', 'tabsService'];
```
Comment (codydaig) [inline modules/core/client/controllers/addTabDialogue.controller.client.js]: Indentation

File: modules/core/client/controllers/adminTabs.controller.client.js [hunk 12]
```diff
@@ -26,6 +28,47 @@ function AdminTabsController (tabsService) {
...
+    })
+    .then(handleSuccess)
+    .catch(handleCancel);
+  }
+
+  function handleSuccess () {
+    //TODO - save button handler
+  }
+
+  function handleCancel () {
```
Comment (codydaig) [inline modules/core/client/controllers/adminTabs.controller.client.js:64]: Alphabetical Order

File: modules/core/client/controllers/editTabDialogue.controller.client.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    $mdDialog.hide();
+  }
+
+  function saveTab () {
+    tabsService.updateTab(vm.tab, tab.id)
+      .then(showSuccess)
+      .catch(displayError);
+  }
+
+  function showSuccess () {
```
Comment (codydaig) [inline modules/core/client/controllers/editTabDialogue.controller.client.js]: Alphabetical Order

**Comment by RRDuke** [comment]
LGTM Pending comments

**Comment by codydaig** [comment]
LGTM

**Pull request merged**